File: src/main/java/me/itzisonn_/meazy/parser/pasing_function/expression/ExpressionParsingFunction.java
package me.itzisonn_.meazy.parser.pasing_function.expression;

import me.itzisonn_.meazy.MeazyMain;
import me.itzisonn_.meazy.parser.ParsingContext;
import me.itzisonn_.meazy.parser.ast.expression.Expression;
import me.itzisonn_.meazy.parser.pasing_function.AbstractParsingFunction;
import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;

@NullMarked
public class ExpressionParsingFunction extends AbstractParsingFunction<Expression> {
    public ExpressionParsingFunction() {
        super("expression");
    }

    @Override
    public Expression parse(ParsingContext context, @Nullable Object... extra) {
        return context.getParser().parseAfter(MeazyMain.getDefaultIdentifier("expression"), Expression.class);
    }
}

File: src/main/java/me/itzisonn_/meazy/parser/pasing_function/expression/ExpressionParsingFunction.kt
package me.itzisonn_.meazy.parser.pasing_function.expression

import me.itzisonn_.meazy.parser.Parser
import me.itzisonn_.meazy.parser.ast.expression.Expression
import me.itzisonn_.meazy.parser.pasing_function.ParsingFunction

object ExpressionParsingFunction : ParsingFunction<Expression>("expression") {
    override fun Parser.parse(vararg extra: Any?): Expression {
        return parse(ListCreationExpressionParsingFunction)
    }
}
